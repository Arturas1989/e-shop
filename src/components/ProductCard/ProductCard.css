.product-card, .bottom{
  display: flex;
  flex-direction: column;
  align-items: center;
  box-shadow: var(--product-card-box-shadow);
}

.bottom{
  padding: 15px;
  gap: 10px;
  box-shadow: none;
}

.bottom > p {
  min-height: 200px;
}

.product-card{
  width: var(--featured-product-width);
  border-radius: var(--featured-product-border-radius);
}

.product-card .img-container, .product-card img{
  width: 100%;
  max-height: 200px;
}

.product-card .img-container{
  padding: 10px;
}

.product-card img{
  object-fit: contain;
}

.price-row{
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: center;
}

.price-row > span{
  font-weight: bold;
}

.feature{
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 30px
}

@media screen and (max-width: 855px){
  .product-card{
    width: 100%;
  }
}