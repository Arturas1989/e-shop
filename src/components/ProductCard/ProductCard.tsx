import { BestSeller } from './BestSeller/BestSeller';
import { Rating } from './Rating/Rating';

type ProductCardProps = {
  name?: string;
};

export const ProductCard = ({ name }: ProductCardProps) => {
  return (
    <div className="product-card">
      <BestSeller isBestSeller={true} />
      <img src={`/assets/${name}.png`} alt="headphones" />
      <div className="bottom">
        <h1>headphones</h1>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quibusdam
          magni sunt quidem impedit voluptate praesentium maiores? Recusandae,
          sequi, reiciendis iste in dolorum molestiae odio exercitationem autem
          nulla ad maiores repellendus.
        </p>
        <Rating stars={5} />
        <div className="price-row">
          <span>$99</span>
          <button>
            <span>Add to cart</span>
            <i className="bi bi-plus"></i>
          </button>
        </div>
      </div>
    </div>
  );
};
