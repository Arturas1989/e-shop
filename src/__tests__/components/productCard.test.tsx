import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { ProductCard } from '../../components/ProductCard/ProductCard';

describe('footer components renders', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <ProductCard />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});