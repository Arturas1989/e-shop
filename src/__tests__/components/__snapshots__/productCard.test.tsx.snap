// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard components renders renders correctly 1`] = `
<div
  className="product-card"
>
  <div
    className="best-seller not-visible"
  >
    best seller
  </div>
  <img
    alt="headphones"
    src="/assets/headphones.jpg"
  />
  <div
    className="bottom"
  >
    <h1>
      headphones
    </h1>
    <p>
      this is headphones
    </p>
    <div
      className="rating"
    >
      <i
        className="bi bi-star-fill"
      />
      <i
        className="bi bi-star-fill"
      />
      <i
        className="bi bi-star-fill"
      />
      <i
        className="bi bi-star-fill"
      />
      <i
        className="bi bi-star-fill"
      />
    </div>
    <div
      className="price-row"
    >
      <span>
        $
        99
      </span>
      <button>
        <span>
          Add to cart
        </span>
        <i
          className="bi bi-plus"
        />
      </button>
    </div>
  </div>
</div>
`;
